FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -209,20 +218,33 @@ - (void)handleOauthDiscoveryCompletion:(OIDServiceConfiguration *_Nullable)confi
   [self createUIWindowForLogin];
   // performs authentication request
   [FIRAppDistributionAppDelegatorInterceptor sharedInstance].currentAuthorizationFlow =
-      [OIDAuthState authStateByPresentingAuthorizationRequest:request
-                                     presentingViewController:self.safariHostingViewController
-                                                     callback:^(OIDAuthState *_Nullable authState,
-                                                                NSError *_Nullable error) {
-          self.authState = authState;
-          if(authState) {
-              [FIRAppDistributionAuthPersistence persistAuthState:authState];
-          }
-          self.isTesterSignedIn = self.authState ? YES : NO;
-          completion(error);
-        }];
+      [OIDAuthState
+          authStateByPresentingAuthorizationRequest:request
+                           presentingViewController:self.safariHostingViewController
+                                           callback:^(OIDAuthState *_Nullable authState,
+                                                      NSError *_Nullable error) {
+                                             self.authState = authState;
+
+                                             // Capture errors in persistence but do not bubble them
+                                             // up
+                                             NSError *authPersistenceError;
+                                             if (authState) {
+                                               [FIRAppDistributionAuthPersistence
+                                                   persistAuthState:authState
+                                                              error:&authPersistenceError];
+                                             }
+
+                                             // TODO (schnecle): Log errors in persistence using
+                                             // FIRLogger
+                                             if (authPersistenceError) {
+                                               NSLog(@"Error persisting token to keychain: %@",
+                                                     [error localizedDescription]);
+                                             }
+                                             self.isTesterSignedIn = self.authState ? YES : NO;
+                                             completion(error);
```

Comment:
If there's an `authPersistenceError` but not an `error`, do we want to surface that in this completion handler?

Reply:
You can also apply the same pattern here where you check `if (error)` at the very beginning of this block, and if so, `self.isTesterSignedIn = NO; self.authState = nil; completion(error)`, then do the same thing right after the call to `persistAuthState`

Reply:
For this case, we explicitly do not want to surface this to the user or to the developer because the authState may be valid for this session even though there was an issue with persistence. So we want to log it, but not actually raise it because the action we would want a developer to take in this case are pretty unclear. I think it is worth logging from an audit perspective so that when a developer writes in to us saying that their users have to sign in every time they open the app we can get some signal on where it's going wrong.